api/src/main/java/org/openmrs/api/impl/PatientServiceImpl.java

```diff
@@ -1638,4 +1638,14 @@ public void checkIfPatientIdentifierTypesAreLocked() {
 	public List<PatientIdentifier> getPatientIdentifiersByPatientProgram(PatientProgram patientProgram) {
 		return dao.getPatientIdentifierByProgram(patientProgram);
 	}
+
+	/**
+	 * @see org.openmrs.api.PatientService#getPatientByIdentifier(java.lang.String)
+	 */
+	@Override
+	@Transactional(readOnly = true)
+	public List<Patient> getPatientByIdentifier(String identifier, boolean matchIdentifierExactly) throws APIException {
+		return Context.getPatientService().getPatientByIdentifier(identifier, matchIdentifierExactly);
```

Comment:
Isn't this just calling itself indefinitely?  Shouldn't there be a method added to the PatientDAO interface and be called from here?

Reply:
I added an abstract method(getPatientByIdentifier()) in PatientService interface where of which this is an implementation  to the method since PatientServiceImpl class implements the interfact.

> Shouldn't there be a method added to the PatientDAO interface and be called from here?

Are you suggesting to have the return as 
```js
return dao.getPatientByUuid(identifier);
```
?